### PR TITLE
[nrfconnect] Added support all-clusters-app

### PR DESCRIPTION
**Pull request by markaj-nordic**
#### Problem
`nrfconnect` platform shall support all-clusters-app example, mainly for testing purposes

#### Change overview
* All clusters are populated with default values on 3 endpoints according to the ZAP file corresponding to the all-clusters-app
* Added README.md with description of the application and steps how to use it

#### Testing
* Tested manually with chip-tool running on the PC and nrf52840 DK
* OTA DFU tested with nrf52840 DK and OTA Provider Linux sample.

### CODE REVIEW

**Comment by jmeg-sfy** [comment]
@markaj-nordic seems you have restyling error before anything else

**Review by Damian-Nordic — APPROVED** [review]

**Comment by markaj-nordic** [comment]
@jmeg-sfy sorry, apparently the restyler is lacking rights to modify workflows. I reformatted the changes manually.

File: examples/all-clusters-app/nrfconnect/main/main.cpp [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+#else
+    err = ConnectivityMgr().SetThreadDeviceType(ConnectivityManager::kThreadDeviceType_MinimalEndDevice);
+#endif
+    if (err != CHIP_NO_ERROR)
+    {
+        LOG_ERR("ConnectivityMgr().SetThreadDeviceType() failed");
+        goto exit;
+    }
+
+    ret = AppTask::Instance().StartApp();
```
Comment (bzbarsky-apple) [inline examples/all-clusters-app/nrfconnect/main/main.cpp:78]: This is now racing against the `StartEventLoopTask` bits above with no synchronization, no?

Either `StartEventLoopTask` needs to come after all this other init work, or this code needs to synchronize (via locking or posting events) against the Matter thread.

**Review by bzbarsky-apple — CHANGES_REQUESTED** [review]

File: examples/all-clusters-app/nrfconnect/main/main.cpp (hunk 1 above)
Comment (markaj-nordic) [reply examples/all-clusters-app/nrfconnect/main/main.cpp]: Thanks for the catch. Fixed. 

Maybe we should consider implementing a kind of template method pattern for the initialization process to keep the init steps always in order.

Comment (bzbarsky-apple) [reply examples/all-clusters-app/nrfconnect/main/main.cpp]: The actual behavior can vary by platform.  You don't have to have a specific order as long as you synchronize...

**Review by bzbarsky-apple — APPROVED** [review]

**Pull request merged**